examples/barycentres.pct.py

```diff
@@ -130,18 +130,18 @@ def fit_gp(x: jnp.DeviceArray, y: jnp.DeviceArray) -> dx.MultivariateNormalTri:
 # In GPJax, the predictive distribution of a GP is given by a [Distrax](https://github.com/deepmind/distrax) distribution, making it straightforward to extract the mean vector and covariance matrix of each GP for learning a barycentre. We implement the fixed point scheme given in (3) in the following cell by utilising Jax's `vmap` operator to speed up large matrix operations using broadcasting in `tensordot`.
 
 # %%
-def sqrtm(A: jnp.DeviceArray):
+def sqrtm(A: jax.Array):
     return jnp.real(jsl.sqrtm(A))
 
 
 def wasserstein_barycentres(
-    distributions: tp.List[dx.MultivariateNormalTri], weights: jnp.DeviceArray
+    distributions: tp.List[dx.MultivariateNormalTri], weights: jax.Array
 ):
     covariances = [d.covariance() for d in distributions]
     cov_stack = jnp.stack(covariances)
     stack_sqrt = jax.vmap(sqrtm)(cov_stack)
 
-    def step(covariance_candidate: jnp.DeviceArray, i: jnp.DeviceArray):
+    def step(covariance_candidate: jax.Array):
         inner_term = jax.vmap(sqrtm)(
             jnp.matmul(jnp.matmul(stack_sqrt, covariance_candidate), stack_sqrt)
```

Comment:
On the typing can we use` jaxtyping`'s `Float` and `Array` instead of `jax.Array`? Additionally do we need to `import typing as tp`? Would it be nicer to `from typing import List`?